src/main/java/com/google/devtools/build/lib/syntax/MethodDescriptor.java

```diff
@@ -130,46 +130,51 @@ public Object invoke(Object obj) throws InvocationTargetException, IllegalAccess
   public Object call(Object obj, Object[] args, Location loc, Environment env)
       throws EvalException, InterruptedException {
     Preconditions.checkNotNull(obj);
+    Object result;
     try {
-      Object result = method.invoke(obj, args);
-      if (method.getReturnType().equals(Void.TYPE)) {
-        return Runtime.NONE;
-      }
-      if (result == null) {
-        if (isAllowReturnNones()) {
-          return Runtime.NONE;
-        } else {
-          throw new EvalException(
-              loc,
-              "method invocation returned None, please file a bug report: "
-                  + getName()
-                  + Printer.printAbbreviatedList(ImmutableList.copyOf(args), "(", ", ", ")", null));
-        }
-      }
-      // TODO(bazel-team): get rid of this, by having everyone use the Skylark data structures
-      result = SkylarkType.convertToSkylark(result, method, env);
-      if (result != null && !EvalUtils.isSkylarkAcceptable(result.getClass())) {
-        throw new EvalException(
-            loc,
-            Printer.format(
-                "method '%s' returns an object of invalid type %r", getName(), result.getClass()));
-      }
-      return result;
+      result = method.invoke(obj, args);
     } catch (IllegalAccessException e) {
       // TODO(bazel-team): Print a nice error message. Maybe the method exists
       // and an argument is missing or has the wrong type.
       throw new EvalException(loc, "Method invocation failed: " + e);
-    } catch (InvocationTargetException e) {
-      if (e.getCause() instanceof FuncallExpression.FuncallException) {
-        throw new EvalException(loc, e.getCause().getMessage());
-      } else if (e.getCause() != null) {
-        Throwables.throwIfInstanceOf(e.getCause(), InterruptedException.class);
-        throw new EvalException.EvalExceptionWithJavaCause(loc, e.getCause());
-      } else {
+    } catch (InvocationTargetException x) {
+      Throwable e = x.getCause();
+      if (e == null) {
         // This is unlikely to happen
-        throw new EvalException(loc, "method invocation failed: " + e);
+        throw new EvalException(loc, "method invocation failed: " + x);
+      }
+      Throwables.propagateIfPossible(e, InterruptedException.class);
+      if (e instanceof FuncallExpression.FuncallException) {
+        throw new EvalException(loc, e.getMessage());
+      }
+      if (e instanceof EvalException) {
+        throw ((EvalException) e).ensureLocation(loc);
       }
+      throw new EvalException.EvalExceptionWithJavaCause(loc, e);
+    }
+    if (method.getReturnType().equals(Void.TYPE)) {
+      return Runtime.NONE;
+    }
+    if (result == null) {
+      if (isAllowReturnNones()) {
+        return Runtime.NONE;
+      } else {
+        throw new EvalException(
```

Comment:
Similarly, let's actually call BugReport.sendBugReport with this eval exception, so it can be tracked, at least inside Google.

Reply:
In this case, it actually breaks a test because `BugReport` raises an unchecked exception if it's invoked in a test. (The test expects the `EvalException`.)

Reply:
Which test? Happy to leave the `BugReport` out for now. But either the test is testing actual possible behavior, in which case we should definitely not file a bug report, or the test is testing an impossible situation, in which case we could just delete/change the test. Definitely not necessary for this change, of course.

Reply:
https://source.bazel.build/bazel/+/be86377d1a3a629f132c76333f38a553d6390ead:src/test/java/com/google/devtools/build/lib/syntax/SkylarkEvaluationTest.java;bpv=;bpt=1;l=1472

It's indicating a (Java-level) programming error.